File: 2014:2015/Maklerbuero/CppMaklerbuero/CppMaklerbuero/Classes/Subclasses/House.cpp
//
//  House.cpp
//  CppMaklerbuero
//
//  Created by Michael Köppl on 22/09/14.
//  Copyright (c) 2014 Michael Köppl. All rights reserved.
//

#include "House.h"

House::House(int i,
             int o,
             std::string an,
             bool tbr,
             double p,
             double as,
             bool mf,
             int fc,
             bool c):Objekt(i, o, an, tbr, p, as){
    
    multifamily = mf;
    floor_count = fc;
    cellar = c;
}

/*get*/
bool House::getMultiFam(){
    return multifamily;
}

int House::getFloorCount(){
    return floor_count;
}

bool House::getCellar(){
    return cellar;
}
/*get*/

/*set*/
void House::setMultiFam(bool v){
    multifamily = v;
}

void House::setFloorCount(int v){
    floor_count = v;
}

void House::setCellar(bool v){
    cellar = v;
}
/*set*/

File: 2014:2015/Maklerbuero/DartMaklerbuero/bin/classes/subclasses/House.dart
library House;

import '../Objekt.dart';

class House extends Objekt{
  
  bool multifamily; //house for multiple families (true) or for a single family (false)
  int floor_count; //number of floors
  bool cellar; //does the house have a cellar (true) or not (false)
  
  House(int objnr, String an, bool tbr, double p, double as, bool mf, int fc, bool c):super(objnr, an, tbr, p, as){
    multifamily = mf;
    floor_count = fc;
    cellar = c;
  }
  
  /*GET*/
  bool getMultiFam(){
    return multifamily;
  }
  
  int getFloorCount(){
    return floor_count;
  }
  
  bool getCellar(){
    return cellar;
  }
  /*GET*/
  
  /*SET*/
  void setMultiFam(bool v){
    multifamily = v;
  }
  
  void setFloorCount(int v){
    floor_count = v;
  }
  
  void setCellar(bool v){
    cellar = v;
  }
  /*SET*/
}